lib/IRGen/GenCall.cpp

```diff
@@ -1228,6 +1257,18 @@ void SignatureExpansion::expandExternalSignatureTypes() {
     }
   }
 
+  if (formalIndirectResult) {
+    // If the result is a formal indirect result in SIL, that means that the
+    // Clang function has an explicit output parameter (e.g. it's a C++
+    // constructor). This means:
```

Comment:
This is a foreign function convention; why aren't we taking the attributes mostly from Clang?

Reply:
Hmm. I agree it would make sense that clang should have the correct attributes on this (clang) function. It doesn't look like we override them anywhere and the generated IR doesn't seem to have a `nocapture` or `sret` on any of the constructor functions. The reasoning seems solid but I'm not sure why @martinboehme thought clang might not do the correct thing.

Removing this change seems fine to me. If I understand correctly, that's what you're suggesting as well, John. Does anyone see a reason this change should stay?

Reply:
Hmm maybe @gribozavr remembers/knows. FWIW, if tests pass, let's clean it up.

Reply:
OK. I'm going to remove it.